types/vue.d.ts

```diff
@@ -0,0 +1,34 @@
+/**
+ * Augment the typings of Vue.js
+ */
+
+import Vue = require("vue");
+import VueRouter = require("./index");
+import { Route, RawLocation } from "./index";
+
+declare module "vue/types/vue" {
+  interface Vue {
+    $router: VueRouter;
+    $route: Route;
```

Comment:
Strictly speaking, these properties are optional but I wrote them non-optional intentionally for convenience. 
Because we would always have these properties practically if we use vue-router.